ProfitApp/src/main/java/bsu/fpmi/profit/ProfitCRUD.java

```diff
@@ -0,0 +1,424 @@
+            PreparedStatement preparedStatement = buildEditPreparedStatement(ad, id);
+            if(preparedStatement != null) {
+                preparedStatement.executeUpdate();
+                preparedStatement.close();
+
+                if (ad.getHashTags() != null) {
+                    editHashtags(id, ad);
+                }
+
+                return true;
+            }
+        }catch(SQLException e){
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public boolean deleteAd(String id){
+        try{
+            String sql = "delete from offer where offer_id = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, Integer.parseInt(id));
+            preparedStatement.executeUpdate();
+            return true;
+        }catch(SQLException e){
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private List<String> readTags(String id) throws SQLException {
+        List<String> tags = new ArrayList<>();
+
+        String sql = "select * from hashtags where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        while(rs.next()){
+            tags.add(rs.getString("tag"));
+        }
+        rs.close();
+        preparedStatement.close();
+        return tags;
+    }
+
+    private class ReviewsWrapper{
+        private List<String> reviews;
+        private double rating;
+
+        public List<String> getReviews() {
+            return reviews;
+        }
+
+        public double getRating() {
+            return rating;
+        }
+
+        public void setRating(double rating) {
+            this.rating = rating;
+        }
+
+        public void setReviews(List<String> reviews) {
+            this.reviews = reviews;
+        }
+    }
+
+    private ReviewsWrapper readReviews(String id) throws SQLException {
+        List<String> reviews = new ArrayList<>();
+        ReviewsWrapper rw = new ReviewsWrapper();
+        double rating = 0.0;
+        int amountReviews = 0;
+        String sql = "select * from reviews where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        while(rs.next()){
+            rating += rs.getInt("rating");
+            amountReviews++;
+            reviews.add(rs.getString("review"));
+        }
+        if(amountReviews != 0) {
+            rating /= amountReviews;
+        }
+
+        rw.setRating(rating);
+        rw.setReviews(reviews);
+        return rw;
+    }
+
+    private Ad readOffer(String id, List<String> tags, List<String> reviews, double rating) throws SQLException {
+        String sql = "select * from offer " +
```

Comment:
These strings are natural candidates for constants to be placed at the beginning of the file. I mean, all the immutable SQL statements strings